lib/containers/urls.pm

```diff
@@ -342,4 +299,27 @@ sub get_3rd_party_images {
     return (\@images);
 }
 
+# Return url of specified image as a string. Die when nothing to return.
+# Specify released => 0 or CONTAINERS_UNTESTED_IMAGES=1 for unreleased image.
+sub get_image_url {
+    my (%args) = @_;
+    $args{version} //= get_required_var('VERSION');
+    $args{arch} //= get_required_var('ARCH');
+    $args{distri} //= get_required_var('DISTRI');
+    $args{released} //= get_var('CONTAINERS_UNTESTED_IMAGES') ? 0 : 1;
+    $args{die} //= 1;    # 0 => do not die
+
+    $args{version} =~ s/^Staging:(?<letter>.)$/Tumbleweed/ if is_tumbleweed || is_microos("Tumbleweed");
+
+    my $url = trim(get_var('CONTAINER_IMAGE_TO_TEST'));
+    return $url if ($url);
+
+    if (supports_image_arch($args{distri}, $args{version}, $args{arch})) {
+        return $images_list{$args{distri}}{$args{version}}{released}->($args{arch}) if ($args{released});
+        return $images_list{$args{distri}}{$args{version}}{totest}->($args{arch});
+    }
+
+    ($args{die}) ? die("Unknown combination of distro/arch or missing CONTAINER_IMAGE_TO_TEST variable.") : return undef;
```

Comment:
It's time to change this error message. To me, when I hit this error in openQA, it was difficult to understand why and I needed to dig into the code. So, for someone who is not used to this code, it's a bit tricky to understand. 

Also, ternary operators are awesome, but when the line is too long, it makes the code a bit unreadable.
So, either you use if-else clause, or store the message in a variable.
Suggestion:
```
my $error_msg = "Cannot find container image for $args{distri} $args{version} $args{arch} or missing CONTAINER_IMAGE_TO_TEST variable";

($args{die}) ? die($error_msg ) : return undef;
```